extension/encoding/googlecloudlogentryencodingextension/internal/dnslog/parser.go

```diff
@@ -0,0 +1,157 @@
+	// gcpDNSHealthyIPs holds addresses in the ResourceRecordSet that are known to be HEALTHY.
+	gcpDNSHealthyIPs = "gcp.dns.healthy_ips"
+	// gcpDNSUnhealthyIPs holds addresses in the ResourceRecordSet that are known to be UNHEALTHY.
+	gcpDNSUnhealthyIPs = "gcp.dns.unhealthy_ips"
+
+	// DNS feature related attributes
+	// gcpDNSDNS64Translated indicates whether DNS64 translation was applied
+	gcpDNSDNS64Translated = "gcp.dns.dns64_translated"
+
+	// VM instance related attributes
+	// gcpDNSVMProjectID holds the Google Cloud project ID of the network from which the query was sent
+	gcpDNSVMProjectID = "gcp.dns.vm.project_id"
+)
+
+type dnslog struct {
+	AliasQueryResponseCode string   `json:"alias_query_response_code"`
+	AuthAnswer             *bool    `json:"authAnswer"`
+	DestinationIP          string   `json:"destinationIP"`
+	DNS64Translated        *bool    `json:"dns64Translated"`
+	EgressError            string   `json:"egressError"`
+	HealthyIps             string   `json:"healthyIps"`
+	Location               string   `json:"location"`
+	Protocol               string   `json:"protocol"`
+	ProjectID              string   `json:"project_id"`
+	QueryName              string   `json:"queryName"`
+	QueryType              string   `json:"queryType"`
+	Rdata                  string   `json:"rdata"`
+	ResponseCode           string   `json:"responseCode"`
+	ServerLatency          *float64 `json:"serverLatency"`
+	SourceIP               string   `json:"sourceIP"`
+	SourceNetwork          string   `json:"sourceNetwork"`
+	SourceType             string   `json:"source_type"`
+	TargetName             string   `json:"target_name"`
+	TargetType             string   `json:"target_type"`
+	UnhealthyIps           string   `json:"unhealthyIps"`
+	VMInstanceID           *int64   `json:"vmInstanceId"`
+	VMInstanceName         string   `json:"vmInstanceName"`
+	VMProjectID            string   `json:"vmProjectId"`
+	VMZoneName             string   `json:"vmZoneName"`
+}
+
+func handleQueryAttributes(log *dnslog, attr pcommon.Map) {
+	shared.PutStr(string(semconv.DNSQuestionNameKey), log.QueryName, attr)
+	shared.PutStr(gcpDNSQueryType, log.QueryType, attr) // TBD in SemConv
+}
+
+func handleResponseAttributes(log *dnslog, attr pcommon.Map) {
+	shared.PutStr(gcpDNSResponseCode, log.ResponseCode, attr) // TBD in SemConv
+	shared.PutStr(gcpDNSAliasQueryResponseCode, log.AliasQueryResponseCode, attr)
+	shared.PutBool(gcpDNSAuthAnswer, log.AuthAnswer, attr)
+	shared.PutStr(gcpDNSAnswerData, log.Rdata, attr) // TBD in SemConv
+}
+
+func handleNetworkAttributes(log *dnslog, attr pcommon.Map) {
+	shared.PutStr(string(semconv.ServerAddressKey), log.DestinationIP, attr)
+	shared.PutStr(gcpDNSSourceVPCNetwork, log.SourceNetwork, attr)
+	shared.PutStr(gcpDNSSourceType, log.SourceType, attr)
+	shared.PutStr(string(semconv.ClientAddressKey), log.SourceIP, attr)
+	shared.PutStr(string(semconv.NetworkTransportKey), log.Protocol, attr)
```

Comment:
Can you lowercase the protocol? Just so it looks like in semantic convention https://github.com/open-telemetry/semantic-conventions/blob/ee434970ad400c279f382a891c84bbaf0e087647/docs/rpc/json-rpc.md